src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/TestRulesContext.cs

```diff
@@ -101,8 +112,19 @@ public TestPosConsensusRules(Network network, ILoggerFactory loggerFactory, IDat
         public T RegisterRule<T>() where T : ConsensusRuleBase, new()
```

Comment:
Hi, can't this function go in `ConsensusRuleEngine` ? (Then can remove here and in `TestConsensusRules` above).

Reply:
If the change is too invasive then maybe a shared class as the code is the same here.

Reply:
fixed